### PR TITLE
Keep PositionStream alive when service status has been changed

### PR DESCRIPTION
**Pull request by JDDV**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Current behavior is that the PositionStream is cancelled when the `Location Service` or permission iOS devices is turned off. The behavior in this PR keeps the stream alive when the location service has been turned off and on again in the settings. 

### :arrow_heading_down: What is the current behavior?

Stops giving location updates when the location service has been turned off and on again in the settings.
### :new: What is the new behavior (if this is a feature change)?

Keeps the PositionStream alive when the location service has been turned off and on again in the settings.

### :boom: Does this PR introduce a breaking change?

No

### :memo: Links to relevant issues/docs

Issue #821

### :thinking: Checklist before submitting

- [x] I made sure all projects build.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [x] I updated the relevant documentation.
- [x] I rebased onto current `master`.

### CODE REVIEW

File: geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m [hunk 1]
```diff
@@ -109,7 +109,7 @@ - (void)locationManager:(CLLocationManager *)manager
         return;
     }
     
-    [self stopListening];
+
```
Comment (mvanbeusekom) [inline geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m:113]: nit: let's remove all these empty lines and just keep one.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Currently the build is failing due to the version number in `pubspec.yaml` being set to `version: NEXT`. This is something I didn't anticipate on and I fail it would be better to make sure the build runs correctly. 

@JDDV could you replace `NEXT` with the version to the next major number (this would be 2.0.0) in both the CHANGELOG.md and the pubspec.yaml file for the geolocator_apple package?

**Comment by mvanbeusekom** [comment]
@JDDV code looks good but there is a unit test failing at the moment. Most likely because the stream is no longer cancelled and the unit test still expecting it is.

**Comment by JDDV** [comment]
> @JDDV code looks good but there is a unit test failing at the moment. Most likely because the stream is no longer cancelled and the unit test still expecting it is.

I don't think that's the case here @mvanbeusekom, the test locally on my pc passes every single time.
Somehow the `TimeoutException` doesn't get thrown, I'll close and re-open the PR (to re-run the checks) and see if that fixes it. Elsewise I'll look into it tomorrow.

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Pull request merged**